src/modules/hip/kernel/tensor_mean.hpp

```diff
@@ -0,0 +1,683 @@
+    {
+        Rpp32u tensorPartialSumArrLength = gridDim_x * gridDim_y * gridDim_z * 3;
+        Rpp32s *tensorPartialSumArr;
+        tensorPartialSumArr = reinterpret_cast<Rpp32s*>(handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+        hipMemsetAsync(tensorPartialSumArr, 0, tensorPartialSumArrLength * sizeof(Rpp32s), handle.GetStream());
+        hipLaunchKernelGGL(tensor_sum_pln3_hip,
+                           dim3(gridDim_x, gridDim_y, gridDim_z),
+                           dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                           0,
+                           handle.GetStream(),
+                           srcPtr,
+                           make_uint3(srcDescPtr->strides.nStride, srcDescPtr->strides.cStride, srcDescPtr->strides.hStride),
+                           tensorPartialSumArr,
+                           roiTensorPtrSrc);
+        hipStreamSynchronize(handle.GetStream());
+        hipLaunchKernelGGL(tensor_mean_grid_3channel_result_hip,
+                           dim3(1, 1, gridDim_z),
+                           dim3(1024, 1, 1),
+                           0,
+                           handle.GetStream(),
+                           tensorPartialSumArr,
+                           gridDim_x * gridDim_y,
+                           tensorMeanArr,
+                           roiTensorPtrSrc);
+    }
+    else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC))
+    {
+        Rpp32u tensorPartialSumArrLength = gridDim_x * gridDim_y * gridDim_z * 3;
+        Rpp32s *tensorPartialSumArr;
+        tensorPartialSumArr = reinterpret_cast<Rpp32s*>(handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+        hipMemsetAsync(tensorPartialSumArr, 0, tensorPartialSumArrLength * sizeof(Rpp32s), handle.GetStream());
+        hipLaunchKernelGGL(tensor_sum_pkd3_hip,
+                           dim3(gridDim_x, gridDim_y, gridDim_z),
+                           dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                           0,
+                           handle.GetStream(),
+                           srcPtr,
+                           make_uint2(srcDescPtr->strides.nStride, srcDescPtr->strides.hStride),
+                           tensorPartialSumArr,
+                           roiTensorPtrSrc);
+        hipStreamSynchronize(handle.GetStream());
+        hipLaunchKernelGGL(tensor_mean_grid_3channel_result_hip,
+                           dim3(1, 1, gridDim_z),
+                           dim3(1024, 1, 1),
+                           0,
+                           handle.GetStream(),
+                           tensorPartialSumArr,
+                           gridDim_x * gridDim_y,
+                           tensorMeanArr,
+                           roiTensorPtrSrc);
+    }
+
+    return RPP_SUCCESS;
+}
+
+// Handle f16/32 datatype
+template <typename T, typename U>
+RppStatus hip_exec_tensor_mean(T *srcPtr,
```

Comment:
Why can't the executors be templated? There are very minor differences I see between them

Reply:
Added templated executors